File: CS/WebSite/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using RoundPanelValidation.Models;

namespace DevExpress.Razor.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        public ActionResult DXControlValidation(){
            return View("DXControlValidation", new Person());
        }
        public ActionResult DXControlValidationPartial([ModelBinder(typeof(DevExpressEditorsBinder))] Person person) {
            return ValidationInternal(person, "SuccessValidation", "DXControlValidationPartial");
        }
        public ActionResult StandardControlValidation() {
            return View("StandardControlValidation", new Person());
        }
        public ActionResult StandardControlValidationPartial([ModelBinder(typeof(DevExpressEditorsBinder))] Person person) {
            return ValidationInternal(person, "SuccessValidation", "StandardControlValidationPartial");
        }

        public ActionResult SuccessValidation() {
            return View("SuccessValidation");
        }

        ActionResult ValidationInternal(Person person, string successAction, string typeAction){
            if (ModelState.IsValid) 
                return View(successAction);

            return View(typeAction, person);

        }
    }
}

File: VB/RoundPanelValidation/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace DevExpress.Razor.Controllers
    Public Class HomeController
        Inherits Controller
        Public Function Index() As ActionResult
            Return View()
        End Function
        Public Function DXControlValidation() As ActionResult
            Return View("DXControlValidation", New Person())
        End Function
        Public Function DXControlValidationPartial(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal person As Person) As ActionResult
            Return ValidationInternal(person, "SuccessValidation", "DXControlValidationPartial")
        End Function
        Public Function StandardControlValidation() As ActionResult
            Return View("StandardControlValidation", New Person())
        End Function
        Public Function StandardControlValidationPartial(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal person As Person) As ActionResult
            Return ValidationInternal(person, "SuccessValidation", "StandardControlValidationPartial")
        End Function

        Public Function SuccessValidation() As ActionResult
            Return View("SuccessValidation")
        End Function

        Private Function ValidationInternal(ByVal person As Person, ByVal successAction As String, ByVal typeAction As String) As ActionResult
            If ModelState.IsValid Then
                Return View(successAction)
            End If

            Return View(typeAction, person)

        End Function
    End Class
End Namespace
